source/binary.cpp

```diff
@@ -571,11 +593,23 @@ spv_result_t Parser::parseOperand(size_t inst_offset,
       convert_operand_endianness = false;
       const char* string =
           reinterpret_cast<const char*>(_.words + _.word_index);
-      size_t string_num_words = (strlen(string) / 4) + 1;  // Account for null.
+      // Compute the length of the string, but make sure we don't run off the
+      // end of the input.
+      const size_t remaining_input_bytes =
+          sizeof(uint32_t) * (_.num_words - _.word_index);
+      const size_t string_num_bytes = strnlen(string, remaining_input_bytes);
```

Comment:
The name is misleading; this will normally be one less than the string's number of bytes.